datafusion/common/Cargo.toml

```
@@ -61,7 +63,7 @@ object_store = { workspace = true, optional = true }
 parquet = { workspace = true, optional = true, default-features = true }
 paste = "1.0.15"
 pyo3 = { version = "0.22.0", optional = true }
-recursive = { workspace = true }
+recursive = { version = "0.1.1", optional = true }
```

```suggestion
recursive = { workspace = true, optional = true }
```

same in other places

---

also, define default `recursive-protection` feature and make it a default feature
add `dep:recursive` to it

---

Added as described thanks